doc/auto_storage.md

```diff
@@ -0,0 +1,663 @@
+resulting in something like this.
+
+```json
+{
+    "search": {
+        "condition": { "sizeGib": 1, "operator": "bigger" }
+    }
+}
+```
+
+Another possible improvement would be to support regular expressions in the string-based form. That
+would make it possible to use searchers like this.
+
+```json
+{ "search": ".*" }
+```
+
+But regular expressions would not play well with libstorage-ng. Since not all device names are
+stored in the devicegraph, it is is necessary to use functions like `find_by_any_name` in order to
+perform an exhaustive search by name.
+
+## Referencing Other Devices
+
+Sometimes is necessary to reference other devices as part of the specification of an LVM volume
+group or RAID. Those can be existing system devices or devices that will be created as response to
+another entry of the Agama profile.
+
+Aliases can be used for that purpose as shown in this example.
+
+```json
+"storage": {
+    "drives": [
+        {
+            "partitions":
+                { "size": "50 GiB", "id": "lvm", "alias": "newPV" }
+        }
+     ],
+     "volumeGroups": [
+        {
+            "name": "newVG",
+            "physicalVolumes": [ "newPV" ],
+            "logicalVolumes": [ { "name": "data", "size": "20 GiB" } ]
+        }
+    ]
+}
+```
+
+If a section that matches several existing devices contains an alias, that alias will be considered
+to be a reference to all the devices. As a consequence, this two examples are equivalent.
+
+```json
+"storage": {
+    "drives": [
+        {
+            "search": {
+                "sort": { "property": "sizeKib", "order": "desc" },
+                "max": 1,
+                "min": 1
+            },
+            "alias": "biggest"
+        },
+        {
+            "search": {
+                "sort": { "property": "sizeKib", "order": "desc" },
+                "max": 1,
+                "min": 1
+            },
+            "alias": "secondBiggest"
+        }
+    ],
+    "mdRaids": [
+        {
+            "devices": [ "biggest", "secondBiggest" ],
+            "level": "raid0"
+        }
+    ]
+}
+
+"storage": {
+    "drives": [
+        {
+            "search": {
+                "sort": { "property": "sizeKib", "order": "desc" },
+                "max": 2,
+                "min": 2
+            },
+            "alias": "big"
+        }
+    ],
+    "mdRaids": [
+        {
+            "devices": [ "big" ],
+            "level": "raid0"
+        }
+    ]
+}
+```
+
+In addition to aliases, a `search` section can appear in all the places in which an alias can be
+used. In that case, the scope of the search is always the whole set of devices in the system (so
+the same conditions can be matched by a disk, a partition, an LVM device, etc.).
+
+```json
+"storage": {
+     "volume_groups": [
+        {
+            "name": "newVG",
+            "physicalVolumes": [
+                { "search": { "condition": { "property": "name", "value": "/dev/sda2" } } }
```

Comment:
For now, I would keep this as a "nice to have". The same can be achieved with aliases, and the scope here in not limited (it would be the only place without limiting the scope).

Reply:
I moved it to the corresponding "Under Discussion" subsection.